File: eform-client/src/app/common/models/eforms/report/eform-report.model.ts
import {EformReportElementModel} from './eform-report-element.model';

export class EformReportModel {
  templateId: number;
  headerImage: string;
  description: string;
  isWorkerNameVisible: boolean;
  isDateVisible: boolean;
  elements: Array<EformReportElementModel> = [];
}

File: eFormAPI/eFormAPI.Web/Infrastructure/Models/Reports/EformReportModel.cs
using System.Collections.Generic;
using eFormAPI.Web.Infrastructure.Database.Entities;

namespace eFormAPI.Web.Infrastructure.Models.Reports
{
    public class EformReportModel
    {
        public int Id { get; set; }
        public int TemplateId { get; set; }
        public string Description { get; set; }
        public string HeaderImage { get; set; }
        public string HeaderVisibility { get; set; }
        public bool IsDateVisible { get; set; }
        public bool IsWorkerNameVisible { get; set; }

        public List<EformReportElementsModel> Elements { get; set; }
            = new List<EformReportElementsModel>();
    }
}